### ISSUE TITLE
3.2.6 Consistent Help - Questions to resolve for consistent testing

### ISSUE THREAD

**Issue by goodwitch** [w3c/wcag#2068]
I love the new proposed SC 3.2.6 for Consistent Help and can't wait to start testing for this in our client work.

As I was reviewing our proposed Deque test methodology for SC 3.2.6 Consistent Help I realized that a few words in the normative of this SC (as it is written now) could be problematic.  

### Problem:
The phrase "same relative order" is hard to apply to a single help mechanism...if we don't define relative to what.

### Proposed Solution:

<ul>
<li><strong>Enhance Normative Text for Clarity:</strong>
<ul>
<li><strong>Current Text:</strong> Success Criterion 3.2.6 Consistent Help (Level A): For each web page within a set of web pages that provides one or more of the following ways of finding help, access to at least one form of help is included in the same relative order on each page
<li><strong>Proposed Test:</strong> Success Criterion 3.2.6 Consistent Help (Level A): For each web page within a set of web pages that provides one or more of the following ways of finding help, access to at least one form of help is included in the same relative order in relation to the page structure on each page
</ul>

<li><strong>Add link to "same relative order" glossary definition in normative of 3.2.6.</strong>
Have you considered the implications of this "Note" in the glossary definition of "same relative order" on 3.2.6?
"Note:  Items are considered to be in the same relative order even if other items are inserted or removed from the original order."  does this mean that if a website had these 3 pages...they would meet 3.2.6:
<ol>
<li>Page 1:
<ul>
  <li>                   MAIN
<ol>
  <li>                      Link 1
  <li>                      Link 2
  <li>                     Content Video
  <li>                     <strong>Help Mechanism</strong>
  <li>                     Link A
</ol>
  <li>                 END-MAIN
</ul>
<li>Page 2:
<ul>
<li>                    MAIN
<ol>
<li>                        Link 1
<li>                        Link 2
<li>                        Content Text
<li>                        Content Image
<li>                        <strong>Help Mechanism</strong>
<li>                        Link A
</ol>
<li>                    END-MAIN
</ul>
<li>Page 3:
<ul>
<li>                    MAIN
<ol>
<li>                        Link 1
<li>                        Link 2
<li>                        Content Text
<li>                        Link 3
<li>                        <strong>Help Mechanism</strong>
<li>                        Link 4
<li>                        Link B
</ol>
 <li>                   END-MAIN
</ul>
</ol>
<li><strong>Add examples in the Understanding document of what is meant by "same relative order".</strong>  
For example (non-exhaustive list):
<ul>
<li>   A help mechanism consistently occurs:
<ul>
<li>            at the beginning of the main content on all pages at breakpoint A
<li>            before the search on all pages at breakpoint A
                on some pages there is an extra link between the help mechanism and the search
<li>            visually on a tab on the side of all pages at breakpoint A (and in a consistent place in the DOM)
<li>               Note: any consistent location within page structure across pages will meet this SC. 
</ul>
</ul>
<li><strong>Add a note that consistency within a Responsive Web Design Breakpoint is required</strong>...but that there can be different designs for each breakpoint.
</ul>

**Comment by goodwitch** [comment]
Oh..and real life example of RWD and location of help being different on wide viewport vs narrow viewport...head to https://www.statefarm.com/ and view at a desktop width (say at least 1366 viewport width).  Notice that "Help" is the 12th tab stop (navigating by keyboard).  Visually (if you know about the Skip to Main link) it is the 3 link on the page.  

Now...look at this same page, https://www.statefarm.com/ with a smaller viewport (say 360 width).  Notice that Help has moved into the hamburger menu (it is now the 9th tab stop once you open the hamburger menu).

**Comment by mraccess77** [comment]
I believe the intent was to only compare within a particular variation (The location in a smaller viewport may be different than in a larger viewport but the mechanism or link should be consistent across a set of web pages. ).  Thus, users would not compare a responsive version to a different responsive variation.

**Comment by bruce-usab** [comment]
Thank you @goodwitch for raising this question.  I don't object to continued word-smithing, but as written, I am not of the opinion that your State Farm example fails 3.2.6, and the current version is more plain language.

Also, <q>in relation to the page structure</q> seems to be introducing a new concept.

**Comment by goodwitch** [comment]
@bruce-usab I totally grok what you are saying...that "in relation to the page structure" is a new concept.  

Do you agree that these questions really need to be addressed in the Understanding doc (for the sake of consistent testing around the world)?

**Comment by alastc** [comment]
> The phrase "same relative order" is hard to apply to a single help mechanism...if we don't define relative to what.

The "same relative order" was based on [Consistent Navigation](https://www.w3.org/TR/WCAG21/#consistent-navigation), has it been a problem for consistent nav?
We did have a long discussion about having it in a 'consistent location', but that gets so hard to define/test, we dropped back to the 'same relative order' concept.

The examples are interesting, and given the definition+note I think they'd pass. However, I'd be surprised if many (any?) people tried to accomplish this with content in the `<main>`. Just from observing where people put links/phone-numbers, it is generally in the header/footer areas.

E.g. [Argos](https://www.argos.co.uk/) (ecommerce) has a help link in the top-right, and [gov.uk](https://gov.uk) has a contact link in the footer. Once in the template it's easy to pass.

Regarding the responsive aspect, the understanding doc includes:
"The location in a smaller viewport may be different than in a larger viewport but the mechanism or link should be consistent across a set of web pages. The location should remain consistent both visually and programmatically."

Not great writing, but it indicates intent. How about:
"The location in a smaller viewport may be different than in a larger viewport **but for each page variation** the mechanism or link should be consistent across a set of web pages. The location should remain consistent both visually and programmatically."

The examples in the understanding could do with some work, I think they are from a previous iteration. How about:

<hr>

-  On-line job application - Some of the application questions may be hard for new job seekers to understand even after reading the contextual help. For example, the form may request their identification number, but they may have several and not know which one to enter. A phone number consistently located near the top of the page will enable them to call for an answer to their question.
- Medical appointment scheduling form - When the service a patient is trying to book is not easily findable within the interface, they may need human help. A built-in messaging option (chat client) enables them to quickly interact with a staff person that can help, without requiring them to manage a second interface.
- Finding a specific policy or procedure - An employee who needs to complete a work task may have difficulty locating the specific policy or procedure document on their employer’s Web site. A "How Do I" page may include the information that enables them to independently complete this task.

**Comment by melaniephilipp** [comment]
Multiple responsive layouts aside... Glenda commented "The phrase "same relative order" is hard to apply to a single help mechanism...if we don't define relative to what." If there is a single help mechanism on a page, what relative order are we talking about? Relative to what? This is different from Consistent Navigation where the relative order within the group of nav elements being assessed.

**Comment by bruce-usab** [comment]
I am all for adding clarity to Understanding.

We have [same relative order](https://www.w3.org/TR/WCAG20/#samerelorderdef) in [2.0 SC 3.2.3](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#consistent-behavior-consistent-locations) and presumably we can reuse that for 3.2.6.

**Comment by goodwitch** [comment]
@alastc and @bruce-usab the difference with using "same relative order" in consistent nav...is that the "relative order" is for the navigation items (more than one nav item).

While in consistent help...the "Help" is a single thing.  So relative order to what?  What if the "help" is a phone number (just text) and is not in the main nav, header of footer.

Are y'all saying that "Help" must be in a consistent NAV, HEADER, FOOTER, SIDEBAR kind of location?  And Help won't qualify as consistent if it is in the MAIN?  

We are drafting our Deque test method (for objective consistent testing).....and that is why I'm getting nitty gritty on this.  We've got lots of folks who do testing...so we have to write a test method that ensures the same test result no matter who does the testing.   "Same relative order" for "Help"...we need to explain "relative to what".   Nav?  Content? Page Structure? Physical visual location (+ DOM order)?

**Comment by alastc** [comment]
I think my last comment leads to a better explanation of 'relative order' for this SC. Where it currently says:

> Regardless of which approach is used, the mechanism must be located in the same relative order on each page within the set of pages. 

We could add:
"When testing this SC, it is the help item which is relative to the rest of the content. When looking at a page, other content that present across the set of web pages and is before the help item should be before the help item on this page. Items which are after the help item on other pages should be after the help item on this page."

It's getting wordy, suggestions welcome.
